libparsec/crates/types/src/certif.rs

```diff
@@ -1151,11 +1225,29 @@ pub enum UnsecureAnyCertificate {
     SequesterAuthority(UnsecureSequesterAuthorityCertificate),
 }
 
+impl_base_load!(AnyCertificate);
+
 impl AnyCertificate {
+    fn check_data_integrity(&self) -> DataResult<()> {
+        match self {
+            AnyCertificate::User(c) => c.check_data_integrity(),
```

Comment:
If this is the reason `check_data_integrity` is declared as `pub(super)`, not that private fields can be accessed from within the module the structure belong to:

```
struct Foo{
    a: usize  // Not pub
}

impl Foo {
    fn b(self) -> usize { self.a * 2 }  // Also not pub
}

fn main() {
    let f = Foo { a: 1 };
    assert_eq!(f.b(), 2);
}
```

Reply:
[6731fa0](https://github.com/Scille/parsec-cloud/pull/7431/commits/6731fa051bb816f0bb76b1ff21d27139208df024)